frontend/src/entities/dataset/dataset-drag-and-drop.tsx

```diff
@@ -0,0 +1,152 @@
+     * It handles both individual files and directories, reading the
+     * content of directories and returning all the files in the directory hierarchy.
+     *
+     * @param dataTransfer - The DataTransfer object containing the dragged files.
+     * @returns A Promise that resolves to an array of File objects.
+     */
+    async function getFilesAsync (dataTransfer: DataTransfer) {
+        const files: File[] = [];
+        for (let i = 0; i < dataTransfer.items.length; i++) {
+            const item = dataTransfer.items[i];
+            if (item.kind === 'file') {
+                if (typeof item.webkitGetAsEntry === 'function') {
+                    const entry = item.webkitGetAsEntry();
+                    const entryContent = await readEntryContentAsync(entry);
+                    files.push(...entryContent);
+                    continue;
+                }
+    
+                const file = item.getAsFile();
+                if (file) {
+                    files.push(file);
+                }
+            }
+        }
+        return files;
+    }
+
+    // Returns a promise with all the files of the directory hierarchy
+    function readEntryContentAsync (entry: FileSystemEntry) {
+        return new Promise<File[]>((resolve) => {
+            let reading = 0;
+            const contents: File[] = [];
+
+            readEntry(entry);
+
+            function readEntry (entry: FileSystemEntry) {
+                if (isFile(entry)) {
+                    reading++;
+                    entry.file((file) => {
+                        reading--;
+                        contents.push(file);
+
+                        if (reading === 0) {
+                            resolve(contents);
+                        }
+                    });
+                } else if (isDirectory(entry)) {
+                    readReaderContent(entry.createReader());
+                }
+            }
+
+            function readReaderContent (reader: FileSystemDirectoryReader) {
+                reading++;
+
+                reader.readEntries(function (entries) {
+                    reading--;
+                    for (const entry of entries) {
+                        readEntry(entry);
+                    }
+
+                    if (reading === 0) {
+                        resolve(contents);
+                    }
+                });
+            }
+        });
+    }
+
+    // Helps with explicit typing
+    function isDirectory (entry: FileSystemEntry): entry is FileSystemDirectoryEntry {
+        return entry.isDirectory;
+    }
+
+    // Helps with explicit typing
+    function isFile (entry: FileSystemEntry): entry is FileSystemFileEntry {
+        return entry.isFile;
+    }
+
+    return (
+        <>
+            <div className='dropzone display-none' onDrop={handleDrop} onDragOver={dragOverHandler} onDragEnter={dragEnterHandler} onDragLeave={dragLeaveHandler}/>
+        </>
```

Comment:
I don't think there is any reason to wrap the div in `<>`

Reply:
Also would it make more sense to [use a ref](https://react.dev/learn/manipulating-the-dom-with-refs#) instead of doing `document.getElementsByClassName('dropzone')[0]`?